test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -192,6 +192,30 @@
 @LuceneTestCase.SuppressReproduceLine
 public abstract class ESTestCase extends LuceneTestCase {
 
+    // A temporary list of ignored timezones.
+    //
+    // These should be cleaned up when all tested jdks (oracle, adoptopenjdk, openjdk etc) have the timezone db included
+    // see when a timezone was included in jdk version here https://www.oracle.com/java/technologies/tzdata-versions.html
+    // In some cases Joda and the JDK timezone data might also be out of sync,
+    // see https://github.com/elastic/elasticsearch/issues/82356
+    protected static final Set<String> IGNORED_TIMEZONE_IDS = Collections.unmodifiableSet(
+        new HashSet<>(
+            Arrays.asList(
+                "Eire",
+                "Europe/Dublin", // dublin timezone in joda does not account for DST
+                "Asia/Qostanay", // part of tzdata2018h
+                "America/Godthab", // part of tzdata2020a (maps to America/Nuuk)
+                "America/Nuuk", // part of tzdata2020a
+                "America/Ciudad_Juarez", // part of tzdata2022g
+                "America/Pangnirtung", // part of tzdata2022g
+                "Europe/Kyiv", // part of tzdata2022c,
+                "Pacific/Kanton", // part of tzdata2021b
+                "Pacific/Niue",
```

Comment:
(in a followup) it might be good to sort this by the tzdata update. Then maybe we can look at each update and add here accordingly. Seems likely we are still missing some.

Reply:
👍 I'll keep an eye on it and will adjust next time. I've already gone over the updates once to make sure everything that was renamed / removed or added is included. Unfortunately some zones seem to be missing in the notes.